tests/test_airbrakes.py

```diff
@@ -1,22 +1,95 @@
+import time
+
 import pytest
 
 from airbrakes.airbrakes import AirbrakesContext
+from airbrakes.imu.data_processor import IMUDataProcessor
+from airbrakes.state import StandByState
 
 
 @pytest.fixture
-def airbrakes_context(imu, logger, servo):
+def airbrakes(imu, logger, servo):
     return AirbrakesContext(logger, servo, imu)
 
 
+@pytest.mark.filterwarnings("ignore:To reduce servo jitter")  # ignore warning about servo jitter
 class TestAirbrakesContext:
     """Tests the AirbrakesContext class"""
 
-    def test_slots(self, airbrakes_context):
-        inst = airbrakes_context
+    def test_slots(self, airbrakes):
+        inst = airbrakes
         for attr in inst.__slots__:
             assert getattr(inst, attr, "err") != "err", f"got extra slot '{attr}'"
 
-    def test_init(self, airbrakes_context, logger, imu, servo):
-        assert airbrakes_context.logger == logger
-        assert airbrakes_context.servo == servo
-        assert airbrakes_context.imu == imu
+    def test_init(self, airbrakes, logger, imu, servo):
+        assert airbrakes.logger == logger
+        assert airbrakes.servo == servo
+        assert airbrakes.imu == imu
+        assert airbrakes.current_extension == 0.0
+        assert isinstance(airbrakes.data_processor, IMUDataProcessor)
+        assert isinstance(airbrakes.state, StandByState)
+        assert not airbrakes.shutdown_requested
+
+    def test_set_extension(self, airbrakes):
+        # Hardcoded calculated values, based on MIN_EXTENSION and MAX_EXTENSION in constants.py
+        airbrakes.set_airbrake_extension(0.5)
+        # TODO: airbrakes.current_extension must be set to 0.5 !!
```

Comment:
Currently, `airbrakes.update()` sets `airbrakes.current_extension` to `airbrakes.servo.current_extension`, which can make the data in the logs look confusing. I propose that the value for `airbrakes.current_extension` should be detached from `servo.current_extension`.

Reply:
Yes you are right. Make it so airbrakes.set_extension() sets both the servo extension and current_extension?